### PR TITLE
fix: `Forge::dropColumn()` always returns `false` on SQLite3 driver

### LINKED ISSUE #8849: Bug(SQLite3): Forge::dropColumn() seems to always return false
### PHP Version

8.3

### CodeIgniter4 Version

4.5.1

### CodeIgniter4 Installation Method

Composer (as dependency to an existing project)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

cli

### Database

SQLite3

### What happened?

I was checking whether dropColumn of forge returned false when migrating down. Which always seemed to be the case.
When checking the implementation I noticed that Forge expects from the driver that `_alterTable` returns a SQL statement to be executed, but in this case SQLite3 will return an empty string, because it runs multiple statements in the background, creating a new temporary table without the specified columns, moving data over and renaming afterwards.

### Steps to Reproduce

```php
<?php

declare(strict_types=1);

namespace App\Database\Migrations;

use CodeIgniter\Database\Exceptions\DatabaseException;
use CodeIgniter\Database\Migration;

class AddAColumnToExistingTable extends Migration
{
    protected string $table = 'existing_table';

    public function up(): void
    {
        $this->forge->addColumn($this->table, [
            'column'   => ['type' => 'varchar', 'null' => false],
        ]) or throw new DatabaseException("Could not add columns to table {$this->table}: " . var_export($this->db->error(), true));
    }

    public function down(): void
    {
        $this->forge->dropColumn($this->table, 'column')
            or throw new DatabaseException("Could drop column from table {$this->table}: " . var_export($this->db->error(), true));
    }
}
```

### Expected Output

None, but the DatabaseException is thrown in the down() method, telling me that no error occured.
But dropColumn() returned false, because the statement is empty.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
Fixes #8849 
Replaces and closes #8931

Changes are coming from the original PR. Not sure if there's anymore needed for this fix.

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/SQLite3/Forge.php [hunk 1]
```diff
@@ -110,6 +110,31 @@ public function dropDatabase(string $dbName): bool
...
     }
 
+    /**
+     * @param list<string>|string $columnNames
+     *
+     * @throws DatabaseException
+     */
+    public function dropColumn(string $table, $columnNames): bool
+    {
+        $columns = is_array($columnNames) ? $columnNames : array_map(trim(...), explode(',', $columnNames));
```
Comment (michalsn) [inline system/Database/SQLite3/Forge.php:120]: This seems redundant as `Table::dropColum()` seems to already handle this.

File: system/Database/SQLite3/Forge.php [hunk 2]
```diff
@@ -110,6 +110,31 @@ public function dropDatabase(string $dbName): bool
...
+        $result  = (new Table($this->db, $this))
+            ->fromTable($this->db->DBPrefix . $table)
+            ->dropColumn($columns)
+            ->run();
+
+        if (! $result && $this->db->DBDebug) {
+            throw new DatabaseException(sprintf(
```
Comment (michalsn) [inline system/Database/SQLite3/Forge.php:127]: Shouldn't throwing an exception be moved to `Table::run()` method? Right now only `dropColumn()` will be handled this way and other operations not.

File: system/Database/SQLite3/Forge.php (hunk 1 above)
Comment (paulbalandan) [reply system/Database/SQLite3/Forge.php:120]: Yes, but this is also used in the exception message formatting.

Comment (michalsn) [reply system/Database/SQLite3/Forge.php:120]: Ok, I get your motivation now.

File: system/Database/SQLite3/Forge.php (hunk 2 above)
Comment (michalsn) [reply system/Database/SQLite3/Forge.php:127]: I keep forgetting how specific the Forge class is in SQLite3.

It looks like we can't prepare any generic error message that will fit all `Table::run()` use cases. I think we can leave it as is.

**Review by michalsn — APPROVED** [review]
We will need a changelog entry.

File: system/Database/SQLite3/Forge.php (hunk 2 above)
Comment (michalsn) [reply system/Database/SQLite3/Forge.php:127]: When something goes wrong, we also throw but on a `BaseConnection::query()` level.

**Review by michalsn — APPROVED** [review]

**Review by samsonasik — APPROVED** [review]

**Pull request merged**